libvast/vast/system/terminate.hpp

```diff
@@ -0,0 +1,68 @@
+/******************************************************************************
+ *                    _   _____   __________                                  *
+ *                   | | / / _ | / __/_  __/     Visibility                   *
+ *                   | |/ / __ |_\ \  / /          Across                     *
+ *                   |___/_/ |_/___/ /_/       Space and Time                 *
+ *                                                                            *
+ * This file is part of VAST. It is subject to the license terms in the       *
+ * LICENSE file found in the top-level directory of this distribution and at  *
+ * http://vast.io/license. No part of VAST, including this file, may be       *
+ * copied, modified, propagated, or distributed except according to the terms *
+ * contained in the LICENSE file.                                             *
+ ******************************************************************************/
+
+#pragma once
+
+#include "vast/defaults.hpp"
+#include "vast/fwd.hpp"
+#include "vast/logger.hpp"
+#include "vast/system/terminator.hpp"
+
+#include <caf/actor_cast.hpp>
+#include <caf/event_based_actor.hpp>
+#include <caf/fwd.hpp>
+#include <caf/scoped_actor.hpp>
+#include <caf/stateful_actor.hpp>
+
+#include <vector>
+
+namespace vast::policy {
+
+struct sequential;
+struct parallel;
+
+} // namespace vast::policy
+
+namespace vast::system {
+
+/// Performs an asynchronous shutdown of a set of dependent actors,
+/// configurable either in sequential or parallel mode of operation. As soon as
+/// all dependent actors have terminated, the returned promise gets fulfilled.
+/// @param self The actor to terminate.
+/// @param xs Owned actors by *self* that need to shutdown prior to *self*.
```

Comment:
The "owning" part seems to be unrelated to this function; here `*self* ` is just an arbitrary actor that is used to dispatch messages. (i.e. one might just as well use this function with a scoped_actor to kill a bunch of other actors in sequence)